### ISSUE TITLE
Heroku deploy - App crashed

### ISSUE THREAD

**Issue by thebravoman** [consul/consul#5029]
### User Story

Managed to deploy on heroku 20 with ruby-2.7.6. Standard 2x machine for $50

### Why is this Needed?
To use it

### Description

git push heroku master

### Current Behavior
After deploy when starting it fails

```
2022-11-11T21:45:00.300908+00:00 heroku[web.1]: Starting process with command `bin/rails server -p ${PORT:-5000} -e production`
2022-11-11T21:45:10.275768+00:00 app[web.1]: => Booting Puma
2022-11-11T21:45:10.275784+00:00 app[web.1]: => Rails 6.0.5.1 application starting in production
2022-11-11T21:45:10.275784+00:00 app[web.1]: => Run `rails server --help` for more startup options
2022-11-11T21:45:18.887590+00:00 heroku[web.1]: Starting process with command `bin/rails server -p ${PORT:-5000} -e production`
2022-11-11T21:45:22.443562+00:00 heroku[web.1]: Process exited with status 0
2022-11-11T21:45:22.207353+00:00 app[web.1]: [4] Puma starting in cluster mode...
2022-11-11T21:45:22.207371+00:00 app[web.1]: [4] * Version 4.3.12 (ruby 2.7.6-p219), codename: Mysterious Traveller
2022-11-11T21:45:22.207371+00:00 app[web.1]: [4] * Min threads: 0, max threads: 16
2022-11-11T21:45:22.207374+00:00 app[web.1]: [4] * Environment: production
2022-11-11T21:45:22.207393+00:00 app[web.1]: [4] * Process workers: 2
2022-11-11T21:45:22.207411+00:00 app[web.1]: [4] * Preloading application
2022-11-11T21:45:22.207673+00:00 app[web.1]: [4] * Listening on tcp://0.0.0.0:4435
2022-11-11T21:45:22.207751+00:00 app[web.1]: [4] * Daemonizing...
2022-11-11T21:45:26.062928+00:00 app[web.1]: => Booting Puma
2022-11-11T21:45:26.062949+00:00 app[web.1]: => Rails 6.0.5.1 application starting in production
2022-11-11T21:45:26.062950+00:00 app[web.1]: => Run `rails server --help` for more startup options
2022-11-11T21:45:37.916059+00:00 app[web.1]: [4] Puma starting in cluster mode...
2022-11-11T21:45:37.916090+00:00 app[web.1]: [4] * Version 4.3.12 (ruby 2.7.6-p219), codename: Mysterious Traveller
2022-11-11T21:45:37.916090+00:00 app[web.1]: [4] * Min threads: 0, max threads: 16
2022-11-11T21:45:37.916091+00:00 app[web.1]: [4] * Environment: production
2022-11-11T21:45:37.916091+00:00 app[web.1]: [4] * Process workers: 2
2022-11-11T21:45:37.916112+00:00 app[web.1]: [4] * Preloading application
2022-11-11T21:45:37.916411+00:00 app[web.1]: [4] * Listening on tcp://0.0.0.0:52276
2022-11-11T21:45:37.916519+00:00 app[web.1]: [4] * Daemonizing...
2022-11-11T21:45:38.274215+00:00 heroku[web.1]: Process exited with status 0
2022-11-11T21:45:38.491936+00:00 heroku[web.1]: State changed from starting to crashed
2022-11-11T21:45:38.495774+00:00 heroku[web.1]: State changed from crashed to starting
2022-11-11T21:45:54.365680+00:00 heroku[web.1]: Starting process with command `bin/rails server -p ${PORT:-5000} -e production`
2022-11-11T21:46:01.761921+00:00 app[web.1]: => Booting Puma
2022-11-11T21:46:01.761962+00:00 app[web.1]: => Rails 6.0.5.1 application starting in production
2022-11-11T21:46:01.761963+00:00 app[web.1]: => Run `rails server --help` for more startup options
2022-11-11T21:46:12.077876+00:00 app[web.1]: [4] Puma starting in cluster mode...
2022-11-11T21:46:12.078280+00:00 app[web.1]: [4] * Version 4.3.12 (ruby 2.7.6-p219), codename: Mysterious Traveller
2022-11-11T21:46:12.078320+00:00 app[web.1]: [4] * Min threads: 0, max threads: 16
2022-11-11T21:46:12.078356+00:00 app[web.1]: [4] * Environment: production
2022-11-11T21:46:12.078398+00:00 app[web.1]: [4] * Process workers: 2
2022-11-11T21:46:12.078421+00:00 app[web.1]: [4] * Preloading application
2022-11-11T21:46:12.079351+00:00 app[web.1]: [4] * Listening on tcp://0.0.0.0:48356
2022-11-11T21:46:12.079569+00:00 app[web.1]: [4] * Daemonizing...
2022-11-11T21:46:12.104997+00:00 app[web.1]: Error waiting for process to terminate: No child processes
2022-11-11T21:46:12.272505+00:00 heroku[web.1]: Process exited with status 22
2022-11-11T21:46:12.393560+00:00 heroku[web.1]: State changed from starting to crashed
2022-11-11T21:46:14.076387+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=promeni-consul.herokuapp.com request_id=28926da5-eec1-4233-ba8b-ae82d607df8a fwd="188.254.161.195" dyno= connect= service= status=503 bytes= protocol=http
2022-11-11T21:46:18.933204+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=promeni-consul.herokuapp.com request_id=394f9cf4-ad3c-485f-bd0d-ef2df1133171 fwd="188.254.161.195" dyno= connect= service= status=503 bytes= protocol=http

```

### Expected Behavior
To visit the url address and to see consul working

### Steps to reproduce

git push heroku  master on heroku with heroku 20 stack and ruby-2.7.6

### Definition of Done

Can visit the url and see it working

**Comment by dbx834** [comment]
I have the same problem. Were you able to figure it out?
I'm using Heroku-20 too (Ruby-2.7.6)

I see the same logs as you:

```
...
2022-11-22T14:39:34.860471+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=consul-tiny.herokuapp.com request_id=43605950-01ef-4688-b6d5-8f9d2f4d7581 fwd="59.93.234.255" dyno= connect= service= status=503 bytes= protocol=https
```

**Comment by thebravoman** [comment]
Hi, I wanted to ask for an apology here. I was in a different context and just tried to quickly deploy on heroku, because a lot of other projects can not be deployed standalone. 

But not consul. The consul team has done an amazing job with the deploy on standard linux installation which is great. I love it. I just got an ubuntu and deployed it in 30 minutes using the installer. That thing is crazy. You guys are awesome. 

I will close this as now I think it is irrelavant. We should not even try to deploy it on heroku. Deploying on a simple VPS is so easy with the installer (I chose a Digital Ocean droplet if anyone is interested)

@dbx834  hope this helps.
